### ISSUE TITLE
Build Sorald using GraalVM and compare benchmarks

### ISSUE THREAD

**Issue by algomaster99** [SpoonLabs/sorald#805]
Deeply inspired by the talk "Java is Going to the Moon: Native Images with GraalVM" delivered by @cesarsotovalero, I would like to do benchmark testing on Sorald's executable built using GraalVM. 

Academic reference: [Initialize once, start fast: application initialization at build time](https://dl.acm.org/doi/10.1145/3360610)

P.S. It is high time I start caring about execution time and the performance of tools I write or maintain :)

**Comment by monperrus** [comment]
would love to see the results!

**Comment by algomaster99** [comment]
Found a maven plugin to perform the tasks - https://graalvm.github.io/native-build-tools/latest/maven-plugin.html.

**Comment by algomaster99** [comment]
I had a go at benchmarking Sorald using GraalVM, however, I did not succeed to build a _working_ native executable. I was successfully able to build a native image using the above plugin, but when I execute the executable generated, I get exceptions related to some internal classes of sonarlint. Since this is hard for me to debug, I opened [an issue](https://community.sonarsource.com/t/error-in-building-maven-project-with-graalvm-whose-dependency-is-on-sonarlint-core/68640) on their community forum. Let's wait for their response.
